main.go

```diff
@@ -86,132 +54,188 @@ func main() {
 	}
 	log.Println("Any pending migrations ran successfully")
 
+	ctx, _ = signal.NotifyContext(ctx, os.Interrupt)
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	svc := &Service{
+		cfg: cfg,
+		db:  db,
+		ctx: ctx,
+		wg:  &wg,
+	}
+
+	dbConfig, err := svc.setupDbConfig()
+	if err != nil {
+		log.Fatalf("Failed to setup DB config: %v", err)
+	}
+
+	// TODO: should not re-set config like this
+	cfg.NostrSecretKey = dbConfig.NostrSecretKey
```

Comment:
Why don't we just pass dbConfig to `Service`?

And then `svc.cfg` would be used only for `CookieDomain`, `CookieSecret` and `Port`

And Line 69 would be
```go
dbConfig, err := svc.setupDbConfig(cfg)
```

Reply:
Bumi changed this for encrypting user config, marking resolved for now